test/extended/openstack/loadbalancers.go

```diff
@@ -382,6 +383,152 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb][Serial] The O
+			podName, err := getPodNameThroughUdpLb(svcIp, fmt.Sprintf("%d", svcPort), "hostname")
+			if err != nil {
+				e2e.Logf("No connectivity (as expected) while accessing the LoadBalancer service on try %d: %q", i, err)
+			} else {
+				results[podName]++
+			}
+		}
+		e2e.Logf("Pods accessed after %d UDP requests:\n%v\n", connNumber, results)
+		o.Expect(len(results)).Should(o.Equal(0), "There is connectivity to svc pods when it shouldn't: %s", results)
+		e2e.Logf("No connectivity to the service as expected")
+
+		// Remove the LoadBalancerSourceRanges spec from the service
+		_, err = jig.UpdateService(func(svc *v1.Service) {
+			svc.Spec.LoadBalancerSourceRanges = nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		e2e.Logf("Removed LoadBalancerSourceRanges spec from the service")
+
+		// Wait until allowed_cidrs in the LB listener is updated to 0.0.0.0/0 (all traffic allowed)
+		allowAllAllowedCidrs := []string{"0.0.0.0/0"}
+		e2e.Logf("Expected allowed_cidrs: '%v'", allowAllAllowedCidrs)
+		o.Eventually(func() []string {
+			lbListener, err := octavialisteners.Get(loadBalancerClient, listenerId).Extract()
+			if err != nil {
+				e2e.Logf("Error ocurred: %v, trying next iteration", err)
+				return []string{}
+			}
+			e2e.Logf("Found AllowedCIDRs: %v", lbListener.AllowedCIDRs)
+			return lbListener.AllowedCIDRs
+
+		}, "10s", "1s").Should(o.ConsistOf(allowAllAllowedCidrs), "Didn't find the expected allowed_cidrs '%v'", allowAllAllowedCidrs)
+		e2e.Logf("Found expected allowed_cidrs '%v' in Openstack LoadBalancer Listener '%s'", allowAllAllowedCidrs, listenerId)
+
+		// Wait until ProvisioningStatus in the LB listener is back to ACTIVE after it's been updated
+		activeProvisioningStatus := "ACTIVE"
+		e2e.Logf("Expected ProvisioningStatus: '%s'", activeProvisioningStatus)
+		o.Eventually(func() string {
+			lbListener, err := octavialisteners.Get(loadBalancerClient, listenerId).Extract()
+			if err != nil {
+				e2e.Logf("Error ocurred: %v, trying next iteration", err)
+				return ""
+			}
+			e2e.Logf("Found ProvisioningStatus: %s", lbListener.ProvisioningStatus)
+			return lbListener.ProvisioningStatus
+
+		}, "10s", "1s").Should(o.Equal(activeProvisioningStatus), "Didn't find the expected ProvisioningStatus '%s'", activeProvisioningStatus)
+		e2e.Logf("Found expected ProvisioningStatus '%s' in Openstack LoadBalancer Listener '%s'", activeProvisioningStatus, listenerId)
+
+		// Check there is still only one listener in the LB and that the Id matches with the previously stored one
+		allPages, err = octavialisteners.List(loadBalancerClient, octavialisteners.ListOpts{LoadbalancerID: loadBalancerId}).AllPages()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		listeners, err = octavialisteners.ExtractListeners(allPages)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(listeners).Should(o.HaveLen(1), "More than one listener found for LB '%s': %v", loadBalancerId, listeners)
```

Comment:
Same here, message is misleading when len=0.

Reply:
Changed to:
```
o.Expect(listeners).Should(o.HaveLen(1), "Unexpected number of listeners for LB '%s'", loadBalancerId)
```